haproxy/eventhandler.py

```diff
@@ -57,7 +58,12 @@ def listen_dockercloud_events():
     events.on_open(on_websocket_open)
     events.on_close(on_websocket_close)
     events.on_message(on_cloud_event)
-    events.run_forever()
+    while True:
+        try:
+            events.run_forever()
+        except dockercloud.AuthError as e:
+            logger.info("Auth error: %s, retry in 1 hour" % e)
+            time.sleep(3600)
```

Comment:
@evanp 
no, once a credential is invalid, it is bad forever. Sending a request with an invalid auth once per hour will not get you blocked by the server. However, if we make the container quits immediately, as many users run this image with autorestart, the container may retry with the bad auth many times in a short period of time, and you may get blocked by the server.